src/main/java/org/jfrog/bamboo/task/ArtifactoryTaskBase.java

```diff
@@ -0,0 +1,60 @@
+package org.jfrog.bamboo.task;
+
+import com.atlassian.bamboo.task.*;
+import com.atlassian.plugin.PluginAccessor;
+import org.jetbrains.annotations.NotNull;
+import org.jfrog.bamboo.util.ServerConfigBase;
+import org.jfrog.bamboo.util.Utils;
+import org.jfrog.build.api.util.Log;
+import org.jfrog.build.extractor.usageReport.UsageReporter;
+
+/**
+ * Created by Bar Belity on 02/12/2019.
+ */
+public abstract class ArtifactoryTaskBase implements TaskType {
+    protected PluginAccessor pluginAccessor;
+
+    protected abstract ServerConfigBase getUsageServerConfig();
+
+    protected abstract String getTaskUsageName();
+
+    protected abstract TaskResult runTask(@NotNull TaskContext context) throws TaskException;
+
+    protected abstract boolean initTask(@NotNull TaskContext context) throws TaskException;
+
+    protected abstract Log getLog();
+
+    public TaskResult execute(@NotNull TaskContext context) throws TaskException {
+        // Initialize task.
+        boolean initSuccess = initTask(context);
```

Comment:
It looks like the initTask *initTask* implementations always return true.
It si therefore better to change initTask to return void.